crates/y-sweet-core/src/sync_kv.rs

```diff
@@ -26,12 +26,13 @@ impl SyncKv {
         callback: Callback,
     ) -> Result<Self> {
         let key = format!("{}/data.ysweet", key);
-        let data = if let Some(snapshot) = store.get(&key).await? {
-            tracing::info!(size=?snapshot.len(), "Loaded snapshot");
-            bincode::deserialize(&snapshot)?
-        } else {
-            BTreeMap::new()
-        };
+        let data =
+            if let Some(snapshot) = store.get(&key).await.context("Failed to get from store.")? {
+                tracing::info!(size=?snapshot.len(), "Loaded snapshot");
+                bincode::deserialize(&snapshot).context("Failed to deserialize.")?
```

Comment:
Might be worth logging more context here and above so we can look into which key didn't exist or which key's value failed to deserialize?

Reply:
Gotcha - does that include the key or the value?